src/services/reportCache.js

```diff
@@ -108,27 +109,50 @@ export const cacheCourses = async (objectiveId, activityReportObjectiveId, cours
 };
 
 const cacheTopics = async (objectiveId, activityReportObjectiveId, topics = []) => {
-  const topicIds = topics.map((topic) => {
-    if (!topic.id) {
-      auditLogger.error(`Error saving ARO topics: ${JSON.stringify(topics)} for objectiveId: ${objectiveId} and activityReportObjectiveId: ${activityReportObjectiveId}`);
+  // Find all topics with missing ids
+  const topicsNeedingLookup = topics.filter((t) => !t.id && t.name);
+  auditLogger.info(
+    'Some topics were missing IDs and required a lookup. '
+    + `ObjectiveId: ${objectiveId}, AROId: ${activityReportObjectiveId}, `
+    + `Raw topics: ${JSON.stringify(topicsNeedingLookup)}`,
+  ); let resolvedTopics = [];
+
+  if (topicsNeedingLookup.length > 0) {
+    const topicNames = topicsNeedingLookup.map((t) => t.name);
+    const foundTopics = await Topic.findAll({
+      where: { name: topicNames },
+    });
+
+    // Log any that weren't found
+    const foundNames = new Set(foundTopics.map((t) => t.name));
+    const missing = topicNames.filter((n) => !foundNames.has(n));
+    if (missing.length) {
```

Comment:
I'm wondering how possible it would be for this case to happen.

Reply:
I think the only way we could end up here is if the UI sends a topic name that doesn't exist in the database, but I'm not sure under what circumstance that would happen

Reply:
We’ve seen that our frontend does occasionally send topics without IDs, and my current theory is that it’s due to the initial data loaded from the API not including them (i.e., the backend only returned name).